aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -150,40 +172,40 @@ class DispatchTable final {
    * @return Kernel function pointing to the right kernel for the given arguments.
    */
    const DispatchTableEntry& lookup(const Stack* stack) const {
-     if (C10_LIKELY(dispatch_strategy_.is_valid_)) {
-       TensorTypeId dispatch_key = dispatch_strategy_.get_dispatch_key(stack);
-       auto found = kernels_.lookup(dispatch_key);
-       if (nullptr != found) {
-         return *found;
-       }
+     return kernels_.map<const DispatchTableEntry&>(
+       [&] (const detail::KernelTable_& table) -> const DispatchTableEntry& {
```

Comment:
do you have perf benchmark that all this stuff gets inlined?

also, why do you need fancy map+lambdas instead of a simple if statement? you don't save any lines of code and it makes it harder for people to read and understand the code (e.g. it took me a while to figure out what's going on

Reply:
I don't have a perf benchmark but this is so a small lambda that it would be very weird if it didn't get inlined.

The reason I'm using `either.map(...)` instead of `if (either.is_left()) {...either.left(); } else { either.right(); }` is actually perf considerations. This code is on a hot path for dispatch and `either.left()` asserts `either.is_left()`, which isn't necessary anymore since we already checked that in the if statement. The `map()` avoids the double check.

In most situations, readability is more important than micro optimizations, but for the dispatch hot patch I make the trade off differently.